lightning/src/ln/channel.rs

```diff
@@ -452,6 +457,9 @@ pub(super) struct Channel<Signer: Sign> {
 	funding_tx_confirmed_in: Option<BlockHash>,
 	funding_tx_confirmation_height: u32,
 	short_channel_id: Option<u64>,
+	/// If this channel was created in recent software, this will be set to the height at which
+	/// channel creation was intiated. This is used to close if funding is never broadcasted.
+	channel_creation_height: Option<u32>,
```

Comment:
IIUC, this doesn't need to be an `Option` if we pass in `ChannelManager`'s best_block_height as a parameter to `Channel::read`!